pkg/kubelet/kubelet.go

```diff
@@ -754,7 +754,7 @@ func NewMainKubelet(kubeCfg *kubeletconfiginternal.KubeletConfiguration,
 		klet.containerLogManager = logs.NewStubContainerLogManager()
 	}
 
-	if kubeCfg.ServerTLSBootstrap && kubeDeps.TLSOptions != nil && utilfeature.DefaultFeatureGate.Enabled(features.RotateKubeletServerCertificate) {
```

Comment:
This is incorrect. The feature gate makes the feature available to you, and is on by default in 1.12+.

You must indicate you want the feature by setting the ServerTLSBootstrap kubelet config or flag to true.